xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCUser.java

```diff
@@ -574,6 +575,71 @@ private void processDecliningInvitationMessage(
+    {
+        MUCRoom room = server.hasChatRoom(packet.getTo().getNode())?server.getChatRoom(packet.getTo().getNode()):null;
+        if (room!=null&&(room.getOwners().contains(packet.getFrom().asBareJID())||
+            room.getAdmins().contains(packet.getFrom().asBareJID())||
+            room.getMembers().contains(packet.getFrom().asBareJID()))||
+            server.getSysadmins().contains(packet.getFrom().asBareJID())||
+            XMPPServer.getInstance().getAdmins().contains(packet.getFrom().asBareJID()))
+        {
+            Element query = packet.getChildElement();
+            if (query!=null&&query.getName().equalsIgnoreCase("query"))
+            {
+                if (query.getNamespace()!=null&&query.getNamespace().getURI().equals("http://jabber.org/protocol/muc#admin"))
+                {
+                    Element item = query.element("item");
+                    if (item!=null&&item.attributeValue("affiliation")!=null&&
+                        item.attributeValue("affiliation").equalsIgnoreCase("member"))
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+    
+    private void processMemberListQuery(IQ packet)
+    {
+        //OF-370: take the role of mucroom, because an occupant which is not in the room does not have a preexisting role
+        MUCRole roleToUseHere = server.getChatRoom(packet.getTo().getNode()).getRole();
+        try
+        {
+            roleToUseHere.getChatRoom().getIQAdminHandler().handleIQ(packet, roleToUseHere);
+        }
+        catch ( ForbiddenException e )
+        {
+            Log.debug("Unable to process IQ stanza: sender don't have authorization to perform the request.", e);
+            sendErrorPacket(packet, PacketError.Condition.forbidden, "You don't have authorization to perform this request.");
+        }
+        catch ( ConflictException e )
+        {
+            Log.debug("Unable to process IQ stanza: processing this request would leave the room in an invalid state (eg: without owners).", e);
+            sendErrorPacket(packet, PacketError.Condition.conflict, "Processing this request would leave the room in an invalid state (eg: without owners).");
+        }
+        catch ( NotAllowedException e )
+        {
+            Log.debug("Unable to process IQ stanza: an owner or administrator cannot be banned from the room.", e);
+            sendErrorPacket(packet, PacketError.Condition.not_allowed, "An owner or administrator cannot be banned from the room.");
+        }
+        catch ( CannotBeInvitedException e )
+        {
+            Log.debug("Unable to process IQ stanza: user being invited as a result of being added to a members-only room still does not have permission.", e);
+            sendErrorPacket(packet, PacketError.Condition.not_acceptable, "User being invited as a result of being added to a members-only room still does not have permission.");
```

Comment:
This error message does not make sense in context of processing a request for a member list.

Reply:
removed exception handler because it is not a invite case